apps/react/src/challenges/memory-game/components/organisms/GridDisplay.jsx

```diff
@@ -0,0 +1,73 @@
+import { useEffect, useCallback, useContext } from 'react';
+import './organisms.css';
+import Cell from '../atoms/Cell';
+import { GameContext } from '../../context';
+
+const Grid = (props) => {
+  const style = {
+    gridTemplateRows: `repeat(${props.rows}, 1fr)`,
+    gridTemplateColumns: `repeat(${props.cols}, 1fr)`,
+    width: `${props.width - 20}px`,
+    height: `${((props.width - 20) / props.cols) * props.rows}px`,
+    maxHeight: `${props.rows * 100}px`,
+    maxWidth: `${props.cols * 100}px`,
+  };
+  return (
+    <div className="grid" style={style} onClick={props.onClick}>
+      {props.children}
+    </div>
+  );
+};
+
+export default function GridDisplay({ userGrid, windowWidth }) {
+  const { dispatch, state } = useContext(GameContext);
+  // Fit inside the widnow height
+  const windowHeight = window.innerHeight - 230;
+  if (windowWidth > windowHeight) {
+    windowWidth = windowHeight;
+  }
+
+  useEffect(() => {
+    setTimeout(() => {
+      dispatch({ type: 'setEmptyUserGrid' });
+      dispatch({ type: 'setLevelReadyForClick', payload: true });
+    }, 3000);
```

Comment:
Shouldn't there be a cleanup for timeout

Reply:
Added a cleanup thanks for reviewing